students/ThanhNhan/lesson07/test_html_render.py

```diff
@@ -0,0 +1,459 @@
+"""
+test code for html_render.py
+
+This is just a start -- you will need more tests!
+"""
+
+import io
+import pytest
+
+# import * is often bad form, but makes it easier to test everything in a module.
+from html_render import *
+
+
+# utility function for testing render methods
+# needs to be used in multiple tests, so we write it once here.
+def render_result(element, ind=""):
+    """
+    calls the element's render method, and returns what got rendered as a
+    string
+    """
+    # the StringIO object is a "file-like" object -- something that
+    # provides the methods of a file, but keeps everything in memory
+    # so it can be used to test code that writes to a file, without
+    # having to actually write to disk.
+    outfile = io.StringIO()
+    # this so the tests will work before we tackle indentation
+    if ind:
+        element.render(outfile, ind)
+    else:
+        element.render(outfile)
+    return outfile.getvalue()
+
+########
+# Step 1
+########
+
+def test_init():
+    """
+    This only tests that it can be initialized with and without
+    some content -- but it's a start
+    """
+    # e = Element()
+
+    # e = Element("this is some text")
+
+
+def test_append():
+    """
+    This tests that you can append text
+
+    It doesn't test if it works --
+    that will be covered by the render test later
+    """
+    # e = Element("this is some text")
```

Comment:
this too -- might as well keep it -- or remove entire test. An empty test does not good.